sos/report/plugins/crio.py

```diff
@@ -104,6 +78,44 @@ def _get_crio_list(self, cmd):
                 ret.pop(0)
         return ret
 
+
+class RedHatCRIO(Plugin, RedHatPlugin, UbuntuPlugin):
+
+    packages = ('cri-o', 'cri-tools')
+    services = ('crio',)
+
+
+    def setup(self):
+        super(RedHatCRIO, self).setup()
+
+        self.add_copy_spec([
+            "/etc/containers",
+            "/etc/crio/crio.conf",
+            "/etc/crio/seccomp.json",
+            "/etc/crio/crio.conf.d/",
+            "/etc/systemd/system/cri-o.service",
+            "/etc/sysconfig/crio-*"
+        ])
+
+        self.add_env_var([
+            'HTTP_PROXY',
+            'HTTPS_PROXY',
+            'NO_PROXY',
+            'ALL_PROXY'
+        ])
+
+        self.add_cmd_output([
+            "ls -alhR /etc/cni",
+            "crio config"
+        ])
+
+        # base cri-o installation does not require cri-tools, which is what
+        # supplies the crictl utility
+        self.set_cmd_predicate(SoSPredicate(self, packages=['cri-tools']))
```

Comment:
This would need to be called before the `super().setup()`, otherwise the `crictl` commands this predicate is intended to gate would have already been added to the collection phase.